rust/private/rustc.bzl

```diff
@@ -584,6 +584,41 @@ def collect_inputs(
 
     return compile_inputs, out_dir, build_env_files, build_flags_files, linkstamp_outs, ambiguous_libs
 
+def _collect_ld_library_path(env, rust_toolchain, cc_toolchain):
+    """Construct the value of `LD_LIBRARY_PATH` based on the existing environment and registered toolchains.
+
+    Args:
+        env (dict): A dict of existing environment variables
+        rust_toolchain (rust_toolchain): The registered rust_toolchain
+        cc_toolchain (CcToolchainInfo): A cc toolchain provider to be used.
+
+    Returns:
+        str: The value of `LD_LIBRARY_PATH`.
+    """
+    sep = ";"
+    ld_paths = []
+
+    # Maintain existing values
+    if "LD_LIBRARY_PATH" in env:
+        ld_paths.extend(env["LD_LIBRARY_PATH"].split(sep))
+    if "DYLD_LIBRARY_PATH" in env:
+        ld_paths.extend(env["DYLD_LIBRARY_PATH"].split(sep))
+
+    # Add the rustc sysroot
+    ld_paths.append("{}/lib".format(rust_toolchain.sysroot))
```

Comment:
I'd have guessed that rustc does this by default when it is constructing the linker invocation internally. Is it an option to copypaste linker invocation flags before and after this PR? Which sysroot library/libraries are not loaded properly without this line?

Reply:
`rustc` (and there rest of the tooling) does so relative to where the binaries are. The issue is if your `rust_toolchain` is not backed by a single repository that contains all files in the same directories, then some things go missing. My aim for https://github.com/bazelbuild/rules_rust/pull/1119 was to allow `rust_toolchain` to be built form arbitrary targets and still work. In https://github.com/bazelbuild/rules_rust/pull/1163 I found that unfortunately `rustc` isn't able to find libraries in toolchains built like this.

Can you expand on 
>  Is it an option to copypaste linker invocation flags before and after this PR

Is this Bazel output (subcommand?) or something else?

Reply:
> My aim for #1119 was to allow rust_toolchain to be built form arbitrary targets and still work. In #1163 I found that unfortunately rustc isn't able to find libraries in toolchains built like this.

Do you know what exactly is the issue? It sounds like a reasonable expectation that a Rust sysroot that contains symlinks to random places should still work ? Do you have a reproduction case? Of course on Windows is hard, symlinks are complicated and paths can only bed 260 chars short and files can't be opened by multiple processes. I'd like to understand what exactly was the problem before accepting a workaround. It'd be doubly useful to know because internally we also create a sysroot full of symlinks, and I don't recall having any issues (CC @krasimirgg).

> Is this Bazel output (subcommand?) or something else?
Repro workplace would be the best, obviously :) But linker invocatoin (flags that rustc passes to the linker) or even the Bazel output would be useful.

Reply:
I think https://github.com/bazelbuild/rules_rust/pull/1163 is the repro case you're asking for. I had split `rust_repository_set` into simple `http_archive` repositories for each Rust asset (`rustc`, `llvm-tools`, `rust-std`) which creates the scenario where the paths of things like `rust-lld` do not have certain files physically next to it. When the `Rustc` action then gets to linking, it can't find the necessary files.